README.md

```diff
@@ -1,18 +1,122 @@
+  *  `Proxy` - A proxy implementation 
+
+`L2`
+  * `ETHStaticAToken` - Tokens on Starknet equivalent to each staticAToken on Ethereum mainnet. Contains the same logic for tracking user rewards as staticATokens and has the same `rewards_index`.
+  * `claimable` - tracks users' claimable rewards and current reward index for each `ETHStaticAToken`
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - a very basic ERC20 to represent the rewards on L2
+  *  `proxy` - a generic implementation of a proxy in starknet
+
+## ETHStaticATokens on L2
+
+Natively, on Aave aTokens grow in balance, not in value. To be able to create this kind of model, it is important to wrap them before bridging, converting them in a token that grows in value, not in balance.
+
+ETHStaticATokens are an implementation of the wrapped aTokens that will continuously increase in value on Starknet because they are backed by the increasing staticATokens amounts locked in the bridge contract on Ethereum. The ETHStaticATokens can then be converted back to staticATokens + rewards.
+
+## Bridging staticATokens from L1<>L2
+
+- Transfer L1->L2: 
+
+Users can either bridge their staticAToken to L2 by calling `deposit()` on `TokenBridge`, or deposit the underlying asset of the staticAToken directly by calling `depositUnderlying()`.
+
+Calling `deposit` will result in the following:
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+
+- Transfer L2->L1:
+
+To bridge their staticATokens back to L1, users need to call `initiate_withdraw()` on Starknet `token_bridge`. 
+
+Calling `initiate_withdraw` will result in the following:
+
+- The amount withdraw of ETHStaticAToken will be burned by the bridge
+- A message will be sent to L1 with the L1 token address, the l1 recipient, and the amount
+
+(@TODO: add more info on how we bridge the current rewards index when withdrawing..)
+ 
+
+
+## Synchronisation of rewards on L1 <> L2
+
+The challenge here was to allow users to continue enjoying the same rewards as on L1 by continously updating -whenever is possible- the `rewards_index` of all ETHStaticATokens to match the value of their respective StaticATokens on L1. To achieve that, we have updated the `_beforeBeforeTokenTransfer()` function on the new implementation of staticATokens (`StaticATokensLMNew.sol`) to send a message (through the bridge) with the latest `reward_index` value. 
```

Comment:
I don't think our users need to know the design changes of aave's static tokens. Just how to use the system